File: LeltarSupportMauiApp/Models/Product.cs
using System.Collections.Generic;

namespace LeltarSupportMauiApp.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public decimal Price { get; set; }
        public bool Active { get; set; } = true;

        // Navigation
        public IList<ProductIngredient> ProductIngredients { get; set; } = new List<ProductIngredient>();
        public IList<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}

File: karton-backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'price',
        'active',
    ];

    public function ingredients()
    {
        return $this->belongsToMany(
                Ingredient::class,
                'product_ingredient',
                'product_id',         
                'ingredient_id'       
            )->withPivot('quantity');
    }

    public function orderItems()
    {
        return $this->hasMany(OrderItem::class);
    }
}
